INT-3748 BridgeHandler should not copy request headers

mbogoevici opened the pull request:
- override `shouldCopyRequestHeaders` to return false;
- modify existing tests (not specific to `BridgeHandler`) that rely on the fact that `BridgeHandler` copies headers and creates a new message instance to use a custom test handler with the expected behaviour;

Commit:
INT-3748 BridgeHandler should not copy request headers

- override `shouldCopyRequestHeaders` to return false;
- modify existing tests (not specific to `BridgeHandler`) that rely on the fact that `BridgeHandler` copies headers and creates a new message instance to use a custom test handler with the expected behaviour;

mbogoevici commented on spring-integration-core/src/test/java/org/springframework/integration/message/TestHandlers.java:
```diff
@@ -89,4 +90,10 @@ public final static Object countingCountDownHandler(final AtomicInteger counter,
 		};
 	}
 
+	public static class RequestHeaderCopyingEchoHandler extends AbstractReplyProducingMessageHandler {
```
maybe extract this as a top level class or create a static method to return it? it doesn't fit the general model of the other factory methods (which return POJOs), but could simplify things.

artembilan replied:
I'd move this class to the `MessageBuilderAtConfigTests` as `private` and forget. What is good, it still works for `<bean>`
Plus update, please, Copyright for all affected classes and add yourself to the `@author`.
Thanks

Commit:
INT-3748 Polishing

- moved test RequestHeaderCopyingMessageHandler
- updated author tags and copyrights

mbogoevici commented:
Added some polishing, as per comments

artembilan commented:
LGMT. Merging...

artembilan commented:
Merged after rebase and squesh
